test/unit/adb-emu-commands-specs.js

```diff
@@ -256,6 +256,38 @@ describe('adb emulator commands', () => {
         mocks.adb.verify();
       });
     }));
+    describe("network speed method", withMocks({adb}, (mocks) => {
+      it("should throw exception on invalid speed", async () => {
+        await adb.networkSpeed('light').should.eventually.be.rejectedWith("Invalid network speed");
+        mocks.adb.verify();
+      });
+      let speeds = {
+        'gsm': adb.NETWORK_SPEED.GSM,
+        'scsd': adb.NETWORK_SPEED.SCSD,
+        'gprs': adb.NETWORK_SPEED.GPRS,
+        'edge': adb.NETWORK_SPEED.EDGE,
+        'umts': adb.NETWORK_SPEED.UMTS,
+        'hsdpa': adb.NETWORK_SPEED.HSDPA,
+        'lte': adb.NETWORK_SPEED.LTE,
+        'evdo': adb.NETWORK_SPEED.EVDO,
+        'full': adb.NETWORK_SPEED.FULL
+      };
+      _.keys(speeds).forEach( (s) => {
+        it("should set network speed correctly", async () => {
```

Comment:
I'm not sure whether dynamic testcases generation is supported this way. @imurchie ?
And even if it is then it would be useful to have the corresponding tested option name present in the title